diff -up ffmpegthumbs-21.12.1/ffmpegthumbnailer/moviedecoder.cpp.omv~ ffmpegthumbs-21.12.1/ffmpegthumbnailer/moviedecoder.cpp
--- ffmpegthumbs-21.12.1/ffmpegthumbnailer/moviedecoder.cpp.omv~	2022-01-15 21:48:24.853777311 +0100
+++ ffmpegthumbs-21.12.1/ffmpegthumbnailer/moviedecoder.cpp	2022-01-15 21:49:31.187056054 +0100
@@ -24,7 +24,6 @@ MovieDecoder::MovieDecoder(const QString
         , m_pFormatContext(pavContext)
         , m_pVideoCodecContext(nullptr)
         , m_pVideoCodec(nullptr)
-        , m_pVideoStream(nullptr)
         , m_pFrame(nullptr)
         , m_pFrameBuffer(nullptr)
         , m_pPacket(nullptr)
diff -up ffmpegthumbs-21.12.1/ffmpegthumbnailer/moviedecoder.h.omv~ ffmpegthumbs-21.12.1/ffmpegthumbnailer/moviedecoder.h
--- ffmpegthumbs-21.12.1/ffmpegthumbnailer/moviedecoder.h.omv~	2022-01-15 21:48:34.149816362 +0100
+++ ffmpegthumbs-21.12.1/ffmpegthumbnailer/moviedecoder.h	2022-01-15 21:49:10.642969704 +0100
@@ -57,8 +57,7 @@ private:
     int                     m_VideoStream;
     AVFormatContext*        m_pFormatContext;
     AVCodecContext*         m_pVideoCodecContext;
-    AVCodec*                m_pVideoCodec;
-    AVStream*               m_pVideoStream;
+    const AVCodec*          m_pVideoCodec;
     AVFrame*                m_pFrame;
     quint8*                 m_pFrameBuffer;
     AVPacket*               m_pPacket;
